lib/plugins.py

```diff
@@ -533,11 +556,13 @@ def __str__(self):
 
     def close(self):
         # remove self from hooks
-        for k in dir(self):
-            if k in hook_names:
-                l = hooks.get(k, [])
-                l.remove((self, getattr(self, k)))
-                hooks[k] = l
+        for name, func in self._hooks_i_registered:
+            l = hooks.get(name, [])
+            try: l.remove((self, func))
+            except ValueError: pass  # this should never happen but it pays to be paranoid.
```

Comment:
these 2 lines still counts as 4 lines of code in my mind :-P

Reply:
Are you saying I should reformat it? or?  I'm a C++ guy. It's hard for me to waste so many lines. I can do so.. if you suggest it more clearly.  I'm not sure I understand.. ha ha.

Reply:
Hehe just bugging you, it's fine :D